polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -3645,13 +3699,16 @@ mlir::Value MLIRScanner::GetAddressOfBaseClass(
         auto PT = value.getType().cast<LLVM::LLVMPointerType>();
         mlir::Type ET;
         if (auto ST =
-                PT.getElementType().dyn_cast<mlir::LLVM::LLVMStructType>()) {
+                PT.getElementType().dyn_cast<mlir::LLVM::LLVMStructType>())
           ET = ST.getBody()[fnum];
-        } else {
-          ET = PT.getElementType()
-                   .cast<mlir::LLVM::LLVMArrayType>()
-                   .getElementType();
-        }
+        else if (auto AT =
+                     PT.getElementType().dyn_cast<mlir::LLVM::LLVMArrayType>())
+          ET = AT.getElementType();
+        else if (auto AT =
+                     PT.getElementType().dyn_cast<mlir::sycl::AccessorType>())
+          ET = AT.getBody()[fnum];
+        else
+          llvm_unreachable("not implemented");
```

Comment:
```suggestion
        llvm_unreachable("not implemented");
```

Reply:
They are not equivalent. The suggested code would assert even if  `PT.getElementType()` is `LLVMStructType`, `LLVMStructType`, or `LLVMArrayType`.